### PR TITLE
Web apps snapshot commands

### PR DESCRIPTION
**Pull request by Nking92**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `./scripts/ci/test_static.sh` locally? (`pip install pylint flake8` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli-extensions/blob/master/docs/extension_summary_guidelines.md).

### CODE REVIEW

**Comment by panchagnula** [comment]
@Nking92 let me know if you need help with  CI failures - looks like Python style issues that can be fixed easily.

File: src/webapp/azext_webapp/custom.py [hunk 1]
```diff
@@ -178,3 +181,35 @@ def create_deploy_webapp(cmd, name, location=None, dryrun=False):
...
+        if slot:
+            target_id = target_id + "/slots/" + slot
+        target = SnapshotRecoveryTarget(id=target_id)
+        request = SnapshotRecoveryRequest(False, snapshot_time=time, recovery_target=target, recover_configuration=restore_config)
+        if source_slot:
+            return client.web_apps.recover_slot(source_resource_group, source_name, request, source_slot)
+        else:
+            return client.web_apps.recover(source_resource_group, source_name, request)
+    elif any([source_resource_group, source_name]):
+        raise CLIError('usage error: source-resource_group and source-name must both be specified if one is used')
```
Comment (derekbekoe) [inline src/webapp/azext_webapp/custom.py]: The proper param names are `--source-resource-group` and `--source-name`.
It's not clear what `source-resource_group and source-name` refer to.

Comment (Nking92) [reply src/webapp/azext_webapp/custom.py]: Thanks, that was a typo. I've updated these

**Comment by panchagnula** [comment]
@derekbekoe we are ready to merge this, let me know if you have any other feedback. Thank you.

**Pull request merged**